### PR TITLE
Rename Extended-Frame to Disperse-Frame

### PR DESCRIPTION
**Pull request by martaiborra**
Change name of Extended-Frame to Disperse-Frame in order to be more explicit.

### CODE REVIEW

**Comment by nmoinvaz** [comment]
I think perhaps it should be "sparse".
image

image
image

File: README_DFRAME_FORMAT.rst [hunk 1]
```diff
@@ -33,9 +33,9 @@ Examples
 
 Structure example
 ^^^^^^^^^^^^^^^^^
-As shown below, an eframe of 4 chunks will be composed of a directory with each chunk file and the frame file::
+As shown below, an dframe of 4 chunks will be composed of a directory with each chunk file and the frame file::
```
Comment (FrancescAlted) [inline README_DFRAME_FORMAT.rst:36]: A dframe

File: blosc/dframe.c [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 
 /* Append an existing chunk into an extended frame. */
```
Comment (FrancescAlted) [inline blosc/dframe.c:39]: extended - > disperse

File: blosc/dframe.c [hunk 3]
```diff
@@ -56,7 +56,7 @@ void* eframe_create_chunk(blosc2_frame* frame, uint8_t* chunk, int32_t nchunk, i
 
 
 /*Get chunk from extended frame. */
```
Comment (FrancescAlted) [inline blosc/dframe.c:58]: Ditto.

Perhaps you can do a (supervised) automatic replace for extended - > disperse in the code too.

**Review by FrancescAlted — CHANGES_REQUESTED** [review]

**Comment by FrancescAlted** [comment]
> I think perhaps it should be "sparse".
> image
> Instead of "disperse".
> image
> Sparse is adjective and disperse is verb form.

Ups, we did not pay enough attention to disperse being a verb. Yeah, sparse was our initial choice, but then the meaning 2 (meager) put us off. But perhaps we should only pay attention to meaning 1 and go ahead with sparse. After all @nmoinvaz is a native speaker and he finds it a better choice anyways. @martaiborra @aleixalcacer , thoughts?

**Comment by FrancescAlted** [comment]
On the other hand, if we adopt sparse, we have a name collision with the short name for the 'serialized' frame (`sframe`) already existing in the code (and we need to find yet another name :-). 'distributed' frame can also be a possibility, but it gives me a sense of distribution among different machines. Oh, the joy of giving names, isn't it fun?

File: README_DFRAME_FORMAT.rst (hunk 1 above)
Comment (martaiborra) [reply README_DFRAME_FORMAT.rst]: Fixed in 2426e5ae2a82ab94f3fc05902aeb22df78beb0b3 .

File: blosc/dframe.c (hunk 2 above)
Comment (martaiborra) [reply blosc/dframe.c]: Fixed in 2426e5ae2a82ab94f3fc05902aeb22df78beb0b3 .

File: blosc/dframe.c (hunk 3 above)
Comment (martaiborra) [reply blosc/dframe.c]: Fix in 2426e5ae2a82ab94f3fc05902aeb22df78beb0b3 .

**Comment by nmoinvaz** [comment]
I dont' understand the use of the term _serialized_ in the context of a memory-backed frame. I would think that saying it is serialized doesn't add any value (because all frames use msgpack - if you are considering the use of msgpack as the serialization mechanism).

I think it might be better to remove the use of the existing term _sframe_ which is used for _serialized frame_. Instead functions like `blosc2_frame_from_sframe` can become `blosc2_frame_from_memory` or `blosc2_frame_from_buffer`. It would more nicely go along with `blosc2_frame_from_file`. Use of the variable `sdata` in `blosc2_frame` can just become `data` or `buffer`.

**Comment by FrancescAlted** [comment]
Thanks @nmoinvaz  for your thoughts.  I agree that we need a simplification in the different naming for frames.  I am not even sure that frames need to be public (or at least we should not need to disclose the internal details).  I'll think more about this and will come with a proposal (probably in the form of PR).

@martaiborra until we accept the new proposal, I think we need to freeze this PR.

**Comment by FrancescAlted** [comment]
We discussed this internally, and we decided to go with 'sparse' frames.  Also, we are gonna try to get rid of all references to frames in the API; specifying `blosc2_storage` should be the only thing a user needs to know about.  Also, the user can always go to the frame description documents in case she wants to know more about the internal storage format.

Closing.

**Pull request closed without merging**